scripts/build.sh

```diff
@@ -105,13 +105,15 @@ if [ "$JAVA_HOME" = "" ]; then
     echo "SET JAVA_HOME=$JAVA_HOME"
 fi
 
-# Ensure gcc version is above 4.9.0 and is not 8.3.1 for faiss 1.7.4+ compilation
+# Ensure gcc version is above 4.9.0 and at least 9.0.0 for faiss 1.7.4+ / SIMD Neon support on ARM64 compilation
 # https://github.com/opensearch-project/k-NN/issues/975
+# https://github.com/opensearch-project/k-NN/issues/1138
+# https://github.com/opensearch-project/opensearch-build/issues/4386
 GCC_VERSION=`gcc --version | head -n 1 | cut -d ' ' -f3`
-GCC_REQUIRED_VERSION=4.9.0
+GCC_REQUIRED_VERSION=9.0.0
 COMPARE_VERSION=`echo $GCC_REQUIRED_VERSION $GCC_VERSION | tr ' ' '\n' | sort -V | uniq | head -n 1`
-if [ "$COMPARE_VERSION" != "$GCC_REQUIRED_VERSION" ] || [ "$GCC_VERSION" = "8.3.1" ]; then
-    echo "gcc version on this env is either older than $GCC_REQUIRED_VERSION, or equals 8.3.1, exit 1"
+if [ "$COMPARE_VERSION" != "$GCC_REQUIRED_VERSION" ]; then
```

Comment:
```suggestion
if [ "$COMPARE_VERSION" != "$GCC_REQUIRED_VERSION" ] || [ "$GCC_VERSION" = "9.3.1" ]; then
```

Reply:
This is not needed as your change will basically exit the script if version is ever 9.3.1.
We only want to make sure gcc is above 9.0.0.